dags/pipeline_small.py

```diff
@@ -199,32 +219,36 @@ def remove_unblurred_images(**context):
         volume_mounts=[],
     )
 
-    detect_containers = KubernetesPodOperator(
-        task_id='detect_containers',
-        namespace=AKS_NAMESPACE,
-        image=DETECT_CONTAINER_IMAGE,
-        env_vars=get_generic_vars(),
-        cmds=["python"],
-        arguments=["/app/inference_batch.py",
-                   "--date", DATE,
-                   "--device", "cpu",
-                   "--weights", "model_final.pth"],
-        labels=DAG_LABEL,
-        name=DAG_ID,
-        image_pull_policy="Always",
-        get_logs=True,
-        in_cluster=True,
-        is_delete_operator_pod=True,
-        log_events_on_failure=True,
-        hostnetwork=True,
-        reattach_on_restart=True,
-        dag=dag,
-        startup_timeout_seconds=3600,
-        execution_timeout=timedelta(hours=4),
-        node_selector={"nodetype": AKS_NODE_POOL},
-        volumes=[],
-        volume_mounts=[],
-    )
+    detect_containers_list = []
+    for worker_id in num_workers:
+        detect_containers = KubernetesPodOperator(
+            task_id='detect_containers',
+            namespace=AKS_NAMESPACE,
+            image=DETECT_CONTAINER_IMAGE,
+            env_vars=get_generic_vars(),
+            cmds=["python"],
+            arguments=["/app/inference_batch.py",
+                       "--date", DATE,
+                       "--device", "cpu",
+                       "--weights", "model_final.pth",
+                       "--worker-id", worker_id],
+            labels=DAG_LABEL,
+            name=DAG_ID,
+            image_pull_policy="Always",
+            get_logs=True,
+            in_cluster=True,
+            is_delete_operator_pod=True,
+            log_events_on_failure=True,
+            hostnetwork=True,
+            reattach_on_restart=True,
+            dag=dag,
+            startup_timeout_seconds=3600,
+            execution_timeout=timedelta(hours=4),
+            node_selector={"nodetype": AKS_NODE_POOL},
+            volumes=[],
+            volume_mounts=[],
+        )
+        detect_containers_list.append(detect_containers)
```

Comment:
Ik denk niet dat dit zo werkt. probeer een list te maken met bijvoorbeeld: [detect_containers_1, detect_containers_2, detect_containers_3] bij 3 workers

Reply:
Je moet hier denk ik de `TaskGroup` functionaliteit gebruiken https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/example_task_group.html#start

Dit is nieuw in Airflow 2, ik heb er geen ervaring mee. Kwestie van een beetje proberen denk ik